src/components/screen/Calender/Calender.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  format,
+  getDate,
+  getDay,
+  addMonths,
+  subMonths,
+  getMonth,
+  endOfWeek,
+  eachDayOfInterval,
+  eachWeekOfInterval,
+  startOfMonth,
+  endOfMonth,
+} from 'date-fns'
+import { useState } from 'react'
+import { CalenderProps } from './Calender.types'
+
+const Calendar: React.FC<CalenderProps> = (onChange) => {
+  const [targetMonth, setTargetMonth] = useState(new Date())
+  const [selectedDates, setSelectedDates] = useState<number[]>([])
+
+  const getCalendarArray = (date: Date) => {
+    const sundays = eachWeekOfInterval({
+      start: startOfMonth(date),
+      end: endOfMonth(date),
+    })
+    return sundays.map((sunday) =>
+      eachDayOfInterval({ start: sunday, end: endOfWeek(sunday) }),
+    )
+  }
+  const calendar = getCalendarArray(targetMonth)
+
+  const handleDateClick = (date: Date) => {
+    const unixTime = date.getTime()
+    setSelectedDates((prevDates) => {
+      if (prevDates.includes(unixTime)) {
+        return prevDates.filter((time) => time !== unixTime).sort()
+      } else {
+        return [...prevDates, unixTime].sort()
+      }
+    })
+  }
+
+  const handlePreviousMonth = () => {
+    setTargetMonth((current) => subMonths(current, 1))
+  }
+
+  const handleNextMonth = () => {
+    setTargetMonth((current) => addMonths(current, 1))
+  }
+
+  const handleSubmit = () => {
+    console.log(selectedDates)
+  }
```

Comment:
惜しい！ここでonChangeを使うことで外部からCalenderの値を受け取ることができる！（buttonのonClickみたいに）

```suggestion
  const handleSubmit = () => {
    onChange(selectedDates)
  }
```

その上で、Calender.stories.tsxのDefaultのargにonChangeを与えることで実際に試すことができる

```
export const Default: Story = {
  args: {
    onChange: (selectedDates: number[]) => {
      alert(selectedDates)
    },
  },
}
```

Reply:
修正した
https://github.com/NUTFes/chosei-chan/pull/22/commits/1b712ff272e56b96f5d18380131048b88d68d7f5